### PR TITLE
feat: add pwa support

### PR DESCRIPTION
**Pull request by LeezQ**
image

use sealos like a desktop app , powerd by pwa

### CODE REVIEW

File: desktop/frontend/next.config.js [hunk 1]
```diff
@@ -1,9 +1,16 @@
 // @ts-check
+const runtimeCaching = require('next-pwa/cache');
+
+const withPWA = require('next-pwa')({
+  dest: 'public',
+  runtimeCaching,
+  disable: process.env.NODE_ENV !== 'production'
```
Comment (LeezQ) [inline desktop/frontend/next.config.js:7]: disable in develop env

File: desktop/frontend/next.config.js [hunk 2]
```diff
@@ -19,7 +26,10 @@ const nextConfig = {
   experimental: {
     newNextLinkBehavior: true
     // fallbackNodePolyfills: false
+  },
+  typescript: {
```
Comment (LeezQ) [inline desktop/frontend/next.config.js:30]: 防止 ts 类型错误导致 build 失败

**Comment by zzjin** [comment]
commit using -s

**Comment by fanux** [comment]
git commit -s --amend && git push -f

**Review by fanux — APPROVED** [review]

**Pull request merged**